File: frontend/types/notification.ts
export interface NotificationItem {
  id: string;
  user_id: string;
  issue_id?: string | null;
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
}

export interface NotificationResponse {
  items: NotificationItem[];
  unread_count: number;
}

File: backend/app/schemas/notification.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    issue_id: uuid.UUID | None
    title: str
    message: str
    is_read: bool
    created_at: datetime
